src/v3/src/hooks/useOnSubmit.ts

```diff
@@ -100,8 +101,21 @@ export const useOnSubmit = (): (options: OnSubmitHandlerOptions) => Promise<void
     try {
       const newTransaction = await fn(payload);
       setIdxTransaction(newTransaction);
-      const isClientTransaction = newTransaction.nextStep?.name === currTransaction?.nextStep?.name;
```

Comment:
This needed to change because we cant assume the current transaction and the new transaction will only have the same name when there are errors. Looking at v2, they use this requestDidSucceed field for determining when to show the errors.